File: leetcode/0167-two-sum-ii---input-array-is-sorted.go
package leetcode

func TwoSum(numbers []int, target int) []int {
	low, high := 0, len(numbers)-1

	for low < high {
		sum := numbers[low] + numbers[high]

		if sum == target {
			return []int{low + 1, high + 1}
		} else if sum > target {
			high -= 1
		} else {
			low++
		}
	}

	return nil
}

File: leetcode/0167-two-sum-ii---input-array-is-sorted.ts
function twoSum(numbers: number[], target: number): number[] {
  let low = 0;
  let high = numbers.length - 1;

  while (low < high) {
    const sum = numbers[low] + numbers[high];

    if (sum === target) {
      return [low + 1, high + 1];
    } else if (sum > target) {
      high--;
    } else {
      low++;
    }
  }

  return [-1, -1];
}
